src/webgpu/api/validation/vertex_access.spec.ts

```diff
@@ -1,91 +1,483 @@
+  .desc(
+    `
+In this test we test the parameter validation in setIndexBuffer and setVertexBuffer, and we test
+that implicit parameter used in setIndexBuffer and setVertexBuffer is computed correctly, and use
+draw and drawIndexed to validate the buffer binding is as we expect.
+    - Test that bound range out-of-bounds on the GPUBuffer is catched by set*Buffer.
+    - Exam that implicit offset/size are computed correctly.
+    - Test that validation catch the situation that buffer offset > buffer size.
+    - Test that setVertexBuffer validate the slot is less than maxVertexBuffers.
+    - Test that given/computed zero index/vertex buffer bound size is valid as long as drawing
+      0 index/vertex.
+    - Test setting index and vertex buffer before/after setPipeline make no different.
+
+Related set*Buffer validation rules:
+    - Index buffer
+        - offset is a multiple of indexFormat's byte size.
+        - offset + size less than or equal to buffer.[[size]].
+    - Vertex buffer
+        - slot < this.[[device]].[[limits]].maxVertexBuffers.
+        - offset is a multiple of 4.
+        - offset + size less than or equal to buffer.[[size]].
+`
+  )
+  .unimplemented();
+
+g.test(`buffer_binding_overlap`)
+  .desc(
+    `
+In this test, we test the buffer overlapping, i.e. one GPUBuffer is bound to multiple vertex buffer
+slot or both index buffer and vertex buffer slot, with different offset setting (range completely
+overlap, partially overlap and no renge overlap). All 4 types of draw call are tested to ensure
+that the setting doesn't cause validation error when drawing.
+    - Test overlapping {vertex/vertex,vertex/index} buffers are valid with/out draw.
+    - Test all range overlapping situation for buffers
+    - Validate that calling set*Buffer before/after setPipeline is the same
+`
+  )
+  .unimplemented();
+
+g.test(`needed_buffer_missing`)
+  .desc(
+    `
+In this test we test that any missing buffer for a used slot will cause validation errors when drawing.
+- All (non/indexed, in/direct) draw commands
+    - A needed vertex buffer is not bound
+        - Was bound in another render pass but not the current one
+        - x= all vertex formats
+- Indexed draw commands,
+    - No index buffer is bound
+`
+  )
+  .unimplemented();
+
+g.test(`unused_buffer_bound`)
+  .desc(
+    `
+In this test we test that a small buffer bound to unused buffer slot won't cause validation error.
+- All draw commands,
+  - An unused {index , vertex} buffer with uselessly small range is bound (immediately before draw
+    call)
+`
+  )
+  .unimplemented();
+
+g.test(`last_buffer_setting_take_account`)
+  .desc(
+    `
+In this test we test that only the last setting for a buffer slot take account.
+- All (non/indexed, in/direct) draw commands
+  - setPl, setVB, setIB, draw, {setPl,setVB,setIB,nothing (control)}, then a larger draw that
+    wouldn't have been valid before that
+`
+  )
+  .unimplemented();
+
+g.test(`index_buffer_OOB`)
```

Comment:
index_buffer_OOB and vertex_buffer_OOB probably should go in a new file, like `api/validation/encoding/cmds/render/direct_draw.spec.ts`

Reply:
Done.